src/python/grpcio/grpc/beta/_server_adaptations.py

```diff
@@ -292,31 +293,40 @@ class _GenericRpcHandler(grpc.GenericRpcHandler):
   def __init__(
       self, method_implementations, multi_method_implementation,
       request_deserializers, response_serializers):
-    self._method_implementations = method_implementations
+    self._method_implementations = {}
+    for method_pair in method_implementations:
+      method = _common.fully_qualified_method(method_pair[0], method_pair[1])
+      self._method_implementations[method] = (
+          method_implementations[method_pair])
+    self._request_deserializers = {}
+    request_deserializers = request_deserializers or {}
+    for method_pair in request_deserializers:
+      method = _common.fully_qualified_method(method_pair[0], method_pair[1])
+      self._request_deserializers[method] = (
+          request_deserializers[method_pair])
+    self._response_serializers = {}
+    response_serializers = response_serializers or {}
+    for method_pair in response_serializers:
+      method = _common.fully_qualified_method(method_pair[0], method_pair[1])
+      self._response_serializers[method] = (
+          response_serializers[method_pair])
```

Comment:
Some part of me wishes that Python included some additional standard utility function to remap dictionary keys (like in `itertools`, or something...), but, I guess I don't feel too strongly about us writing one and using it to reduce the code duplication here (and I guess it'd just end up looking a lot like a dictionary-comprehension/generator anyway). Just a thought.